src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/AbstractFileWriter.cs

```diff
@@ -372,6 +372,38 @@ protected static bool IsKeyword(string name)
             }
         }
 
+        protected List<Kind> GetKindsOfFieldOrNearestParent(TreeType nd, Field field)
+        {
+            if (!IsOverride(field) || field.Kinds.Count > 0)
```

Comment:
It feels like we can remove the `!IsOverride(field)` check here, because:

- if the override has kinds we just use them
- if the override doesn't have kinds we will jump to the while loop and look for kinds
- if the node isn't an override or nothing in the hierarchy has kinds, we will return `null` which is treated the same as an empty `Kinds`

Reply:
Please check my thinking here, though, since I could be wrong :smile:

Reply:
> if the node isn't an override or nothing in the hierarchy has kinds, we will return null which is treated the same as an empty Kinds

The field can be a non-override with non-empty kinds that we want to return.
Removing the check will cause the kinds for non-overrides to be lost.

Reply:
Sorry, I think I did not describe the flow very well. The suggested change appears to still preserve the Kinds on non-override fields the way we want.

Reply:
Regardless, whether you decide the suggestion is correct or not, I'm happy to take this fix :)

Reply:
It won't preserve them since it won't go into the loop and will jump to `return null`.

I could change `return null` to `return field.Kinds`, but I'm completely unsure about that. The last line will be reachable when the loop is entered and completes without finding kinds on the hierarchy, hence we could end on some node that we are interested in.

Reply:
It looks like we could remove the initial `if`. Perhaps something like:
```csharp
which ((field.Kinds is null || field.Kinds.Count == 0) && IsOverride(field))
{
    nd = GetTreeType(nd.Base);
    field = nd switch { ... };
}
return field.Kinds;
```

Reply:
I was a little bit afraid to change `return null` at the end to `return field.Kinds` in case I'll be returning wrong kinds, but with a second thought this seems impossible.